microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/metrics/properties/MicroProfileMetricsProvider.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.metrics.properties;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.search.SearchMatch;
+
+import com.redhat.microprofile.jdt.core.AbstractAnnotationTypeReferencePropertiesProvider;
+import com.redhat.microprofile.jdt.core.IPropertiesCollector;
+import com.redhat.microprofile.jdt.core.SearchContext;
+import com.redhat.microprofile.jdt.core.utils.JDTTypeUtils;
+
+/**
+ * Properties provider that provides MicroProfile Metrics properties from the
+ * 
+ * @author David Kwon
+ * 
+ * @see https://github.com/eclipse/microprofile-metrics/blob/dc94b84ec90dd4a0bc983336e4273247a4e415cc/spec/src/main/asciidoc/architecture.adoc
+ *
+ */
+public class MicroProfileMetricsProvider extends AbstractAnnotationTypeReferencePropertiesProvider {
+
+	private static final String MP_METRICS_ADDED = MicroProfileMetricsProvider.class
+			.getName() + "#MicroProfileMetrics";
+	
+	public static final String METRIC_ID = "org.eclipse.microprofile.metrics.MetricID";
+
+	public static final String GLOBAL_TAGS_VARIABLE = "mp.metrics.tags";
+
+	public static final String APPLICATION_NAME_VARIABLE = "mp.metrics.appName";
+	
+	@Override
+	public void collectProperties(SearchMatch match, SearchContext context, IProgressMonitor monitor) {
+		IJavaProject javaProject = context.getJavaProject();
+		if (JDTTypeUtils.findType(javaProject, METRIC_ID) == null) {
+			return;
+		}
+		if (context.get(MP_METRICS_ADDED) == null) {
+			IPropertiesCollector collector = context.getCollector();
+			String docs = "List of tag values.\r\n"
```

Comment:
Use context.getDocumentFormat() to generate markdown (like `) or not..

Reply:
@angelozerr is this still needed? If yes, maybe something like
```
if (no markdown support) {
    docs = docs.replaceAll("`", "");
}
```
to remove the ` from the documentation?

Reply:
For the moment we consider that we return every time markdown, please forget my comment. We will improve that in an another issue.